general/g.mapsets/main.c

```diff
@@ -130,7 +163,27 @@ int main(int argc, char *argv[])
         }
     }
 
-    fs = G_option_to_separator(opt.fs);
+    if (strcmp(opt.format->answer, "json") == 0)
+        format = JSON;
+    else
+        format = PLAIN;
+    if (format == JSON) {
+        fatal_error_option_value_excludes_option(
+            opt.format, opt.fsep, _("Separator is part of the format."));
+    }
+
+    /* the field separator */
+    if (opt.fsep->answer) {
+        fsep = G_option_to_separator(opt.fsep);
+    }
+    else {
+        /* A different separator is needed to for each format and output. */
+        if (format == PLAIN) {
+            fsep = G_store("space");
```

Comment:
I would do
```suggestion
            fsep = G_store(" ");
```
and then you should be able to completely remove the code on lines
https://github.com/OSGeo/grass/pull/2542/files#diff-1e00df26754bc117bfc80971ed1433ff5119fa1b80c3bceddf78725a36843ed8L17-L25
and the same for the other function.